OKR.md

```diff
@@ -0,0 +1,50 @@
+# Quarterly Objectives and Key Results
+
+We try to frame our ongoing work using a process based on quarterly Objectives and Key Results (OKRs). Objectives reflect outcomes that are challenging, but realistic. Results are tangible and measurable.
+
+## 2018 Q4
+
+**go-ipfs handles large datasets (1TB++) without a sweat**
+- `PX` - It takes less than 48 hours to transfer 1TB dataset over Fast Ethernet (100Mbps)
+- `PX` - It takes less than 12 hours to transfer 200GB sharded dataset over Fast Ethernet (100Mbps)
+- `PX` - There is a prototype implementation of GraphSync for go-ipfs
+- `PX` - There is a better and more performant datastore module (e.g Badger or better)
+- `P1` - Rewrite pinning data structures to support large data sets / many files performantly
```

Comment:
I would explicitally talk about adding "alternative pinning strategies". go-ipfs can already pin very large data sets performantly in terms of the pin data structure (the problems is perhaps verifying and traversing and safely gc-ing after unpinning them). Cluster's need are much more specific and are about adding one extra pinning mode. The wider discussion about the pinning system goes beyond the data structures too, and affects multiple components, so a KR about that would need different wording than this too.

Reply:
@Stebalien could you comment about what might be a better phrasing?

Reply:
> (the problems is perhaps verifying and traversing and safely gc-ing after unpinning them). 

Unfortunately, the datastructure doesn't support *many* pins. See: https://github.com/ipfs/go-ipfs/issues/5221

However, I agree. Part of this goal is to make the pinning datastructures *flexible*.

Reply:
Just FYI: Personally, I think we should switch to using inode like structures (with small blocks inline in the inode itself) in the Blockstore.  This will be a major infrastructure change but I think it will make a lot of things a lot easier down the road, including pinning.  I created a placeholder issue for this at https://github.com/ipfs/go-ipfs/issues/5528, I will flush out the details later today or tomorrow.

Reply:
> Unfortunately, the datastructure doesn't support many pins

Agreed, I was thinking of large datasets as a thing with a single root cid and not on the pinset as a whole. But yes, the pinset itself is expensive to edit when it's very large. :+1:

Reply:
One simple way of solving this would be transitioning pinset into datastore (backed by DB) directly instead of using ipfs objects for that.

Reply:
cluster wants the ability to pin partial dags - ok with a phrasing like: 
"rewrite pinning data set to support large pin sets and flexible pin modes"

Reply:
I agree with @Kubuxu on using an actual database but, baring that, we could use the new go-ipld-hamt structure (now that we've finally merged the refmt cbor ipld branch).